build-image/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM golang:1.21.9-bullseye
+FROM golang:1.21.11-bullseye
```

Comment:
Building images locally is a pain. Let's if we can merge https://github.com/cortexproject/cortex/pull/6026 before this PR and make this PR easier

Reply:
@dsabsay ok that was merged, make a separate PR for this Dockerfile change as explained on https://cortexmetrics.io/docs/contributing/how-to-update-the-build-image/